intermediate_source/process_group_cpp_extension_tutorial.rst

```diff
@@ -0,0 +1,284 @@
+Customize Process Group Backends Using Cpp Extensions
+=====================================================
+
+**Author**: `Feng Tian <https://github.com/ftian1>`__, `Shen Li <https://mrshenli.github.io/>`__
+
+
+Prerequisites:
```

Comment:
Should we also add "cpp extensions" as a prerequisite?

Reply:
ah, yes, let me add that